### PR TITLE
Don't crash if GE can't check IsDarkTheme

### LINKED ISSUE #8904: Failed to detect IsDarkTheme
## Current behaviour

I tried to run Git Extensions (master). Few months again, I never had a problem.
Today, Git Extensions crashes at : 

GitUI\Theming\ThemeModule.cs:56

```
IsDarkTheme = theme.SysColorValues[KnownColor.Window].GetBrightness() < 0.5;
```

because ``theme.SysColorValues`` is empty. A simple check if key exists solve the problem.

## Expected behaviour

No crash.

## Steps to reproduce

Just start the application. I don't know why GE crashes.

## Did this work in previous version of GitExtensions

Yes :
- Git Extensions 3.4.3.9999
- Build d4b0f48bbf3e39a71f5d7ff5231be5678d4aa0f1
- Git 2.30.0.windows.1
- Microsoft Windows NT 10.0.19042.0
- .NET Framework 4.8.4300.0
- DPI 120dpi (125% scaling)

## Environment

- GitExtensions version: 2bf1e3c9cc4b9ddbe4b2bdf5d37d03b86c6b45cc
- GIT version: Git 2.30.0.windows.1
- OS version: Microsoft Windows NT 10.0.19042.0
- .NET version: .NET Framework 4.8.4300.0
- DPI 120dpi (125% scaling)

## Diagnostics

``G:\Github\gitextensions\GitUI\Theming\ThemeRepository.cs:27`` with  ``GetTheme`` uses

``themePath = C:\Users\xxx\AppData\Roaming\GitExtensions\GitExtensions\Themes\migrated.css`` :

```
.OtherTag { color: #808080}
.AuthoredHighlight { color: #ffffe0}
.HighlightAllOccurences { color: #ffffe0}
.Tag { color: #00008b}
.Graph { color: #8b0000}
.Branch { color: #8b0000}
.RemoteBranch { color: #008000}
.DiffSection { color: #e6e6e6}
.DiffRemoved { color: #ffc8c8}
.DiffRemovedExtra { color: #ffa5a5}
.DiffAdded { color: #c8ffc8}
.DiffAddedExtra { color: #87ff87}
```

### PR DESCRIPTION
**Pull request by bansan85**
Fixes #8904 

## Proposed changes

- Improve reading file theme.css

### Before

#8904 crashes at startup.

### After

Don't crash.

## Test methodology 

Can't write test without refactoring.

``GitUI.Theming.ThemeModule.Repository`` is private and statically initialized.

To test this class, ThemeModule should be refactored like ``GitUI.Theming.ThemeLoader`` to use interface and add a Load method to customize Repository field.

## Test environment(s) 

- Git Extensions 3.4.3.9999
- Build d4b0f48bbf3e39a71f5d7ff5231be5678d4aa0f1
- Git 2.30.0.windows.1
- Microsoft Windows NT 10.0.19042.0
- .NET Framework 4.8.4300.0
- DPI 120dpi (125% scaling)

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: GitUI/Theming/ThemeModule.cs [hunk 1]
```diff
@@ -83,8 +83,10 @@ private static ThemeSettings LoadThemeSettings()
                 return CreateFallbackSettings(invariantTheme);
             }
 
-            IsDarkTheme = theme.SysColorValues[KnownColor.Window].GetBrightness() < 0.5;
-
+            if (theme.SysColorValues.ContainsKey(KnownColor.Window))
```
Comment (mstv) [inline GitUI/Theming/ThemeModule.cs:86]: Use `TryGetValue`, please.

File: GitUI/Theming/ThemeModule.cs [hunk 2]
```diff
@@ -83,8 +83,10 @@ private static ThemeSettings LoadThemeSettings()
...
+            {
+                IsDarkTheme = theme.SysColorValues[KnownColor.Window].GetBrightness() < 0.5;
+            }
```
Comment (mstv) [inline GitUI/Theming/ThemeModule.cs:89]: Missing newline after `}` let's the build fail.
```suggestion
            }
            
```

**Review by mstv — APPROVED** [review]
Thank you.

**Review by gerhardol — CHANGES_REQUESTED** [review]
According to 
https://github.com/gitextensions/gitextensions/issues/8904#issuecomment-787517984

**Review by gerhardol — APPROVED** [review]

**Comment by RussKie** [comment]
@NikolayXHD is this fix still required with #8907?

**Comment by bansan85** [comment]
@RussKie No. #8907 already have the fix.

**Pull request closed without merging**